### PR TITLE
Support enum value, union type, and bean field deprecation

### PR DESCRIPTION
**Pull request by carterkozak**
## Before this PR
No support for per-field/value deprecation metadata.

## After this PR
==COMMIT_MSG==
Support enum value, union type, and bean field deprecation
==COMMIT_MSG==

https://github.com/palantir/conjure/pull/494

### CODE REVIEW

**Comment by carterkozak** [comment]
This will require the conjure change to merge, I've verified tests pass when I use a local publication.

File: conjure-java-core/build.gradle [hunk 1]
```diff
@@ -45,7 +45,9 @@ dependencies {
     compile 'com.fasterxml.jackson.datatype:jackson-datatype-jdk8'
     compile 'com.google.googlejavaformat:google-java-format'
     compile 'com.google.guava:guava'
-    compile 'com.palantir.conjure:conjure-generator-common'
+    compile 'com.palantir.conjure:conjure-generator-common', {
+        exclude group: 'com.palantir.conjure.java', module: 'conjure-lib'
```
Comment (ferozco) [inline conjure-java-core/build.gradle]: why is this necessary?

Comment (carterkozak) [reply conjure-java-core/build.gradle]: Same reason this is duplicated on line 63, a recent change to conjure-generator-common must have added a new dependency on conjure-lib which breaks gcv write-locks picking between conjure-lib (5.3 from conjure project) and the local project link.

Comment (ferozco) [reply conjure-java-core/build.gradle]: Ok, I think we can make gradle's resolution strategy to prefer project dependencies https://docs.gradle.org/current/dsl/org.gradle.api.artifacts.ResolutionStrategy.html

**Comment by ferozco** [comment]
👍 Thanks carter!

File: conjure-java-core/build.gradle (hunk 1 above)
Comment (carterkozak) [reply conjure-java-core/build.gradle]: Ah fantastic, thanks!
